### PR TITLE
Feat #3595: Multi site administration and database

### PR DESCRIPTION
**Pull request by stu01509**
Feat #3595

#### Short description of what this resolves:
Refine the `admin.php` and `sql_upgrade.php` pages.

#### Changes proposed in this pull request:
admin.php
- Remove internal CSS style and outdated code.
- Remove CSS and Javascript outdated attributes.
- Add the Bootstrap table, button style and tooltip.

image

sql_upgrade.php
- Add the Bootstrap style.
image

### CODE REVIEW

File: admin.php [hunk 1]
```diff
@@ -41,218 +41,191 @@ function sqlQuery($statement, $link)
...
 <body>
-    <div class='container'>
+    <div class='container mt-3'>
         <div class="row">
-            <div class="col-sm-12">
-                <div class="page-header">
-                    <h2>OpenEMR Multi Site Administration <a class="oe-pull-away oe-help-redirect" data-target="#myModal" data-toggle="modal" href="#" id="help-href" name="help-href" style="color:#676666" title="Click to view Help"><i class="fa fa-question-circle" aria-hidden="true"></i></a></h2>
+            <div class="col-12">
+                <div class="page-header d-flex justify-content-between align-items-center">
+                    <h2>OpenEMR Multi Site Administration</h2>
```
Comment (stu01509) [inline admin.php:56]: Hi @tywrenn 

Looks like the `.page-header` the class was deprecated in Bootstrap 4.0
Should I remove all `.page-header` class?

https://getbootstrap.com/docs/4.0/migration/#typography

File: admin.php [hunk 2]
```diff
@@ -41,218 +41,191 @@ function sqlQuery($statement, $link)
...
-                        </div>
-                        <div class="modal-footer" style="margin-top:0px;">
-                           <button class="btn btn-link btn-cancel oe-pull-away" data-dismiss="modal" type="button">Close</button>
-                           <!--<button class="btn btn-secondary btn-print oe-pull-away" data-dismiss="modal" id="print-help-href" type="button">Print</button>-->
+                            <iframe src="" id="targetiframe" class="h-100 w-100" style="overflow-x: hidden; border:none"
+                                allowtransparency="true"></iframe>
                         </div>
+                    <div class="modal-footer mt-0">
+                        <button class="btn btn-secondary btn-cancel" data-dismiss="modal" type="button">Close</button>
                     </div>
```
Comment (stu01509) [inline admin.php:207]: Also, I found the `btn-cancel` our custome font-awesome class didn't work on this page, so should I remove it?

Comment (bradymiller) [reply admin.php:207]: yep, would remove there since this script is runs independent from OpenEMR context and doesn't have access (since it is not able to bring in globals.php).

Comment (stu01509) [reply admin.php]: OK, I removed it and updated the commit  :)

File: sites/default/sqlconf.php [hunk 3]
```diff
@@ -1,30 +1,31 @@
 <?php
 //  OpenEMR
```
Comment (bradymiller) [inline sites/default/sqlconf.php:2]: @stu01509 , will need to revert the changes in this script (note we have all done this at one time or another :) )

Comment (stu01509) [reply sites/default/sqlconf.php]: Thanks for your mentioned, I removed it.

File: admin.php (hunk 1 above)
Comment (bradymiller) [reply admin.php:56]: hi @stu01509 , Willing to bet @tywrenn would want you to remove the `.page-header` class on the scripts as you work on them (rather than trying to change them all at once).

Comment (bradymiller) [reply admin.php:56]: So would remove it here and then use the utility classes instead to make it look nice.

Comment (stu01509) [reply admin.php]: Updated the commit, please review it again.
Also, should I create a PR to remove the `.page-header` class?

**Comment by bradymiller** [comment]
@stu01509 ,
Tested this out. The admin script looks really nice. The upgrade scripts also looks nice, but seem odd that the text is right at the left border:
image

File: admin.php [hunk 4]
```diff
@@ -41,218 +41,191 @@ function sqlQuery($statement, $link)
...
-                    }
+                <div class="table-responsive">
+                    <table class='table table-striped text-center'>
+                        <tr>
+                            <th >Site ID</th>
+                            <th >DB Name</th>
+                            <th >Site Name</th>
+                            <th >Version</th>
+                            <th >Is Current</th>
+                            <th >Log In</th>
```
Comment (bradymiller) [inline admin.php:74]: the whitespace in the `<th >` tags looks odd

File: admin.php [hunk 5]
```diff
@@ -41,218 +41,191 @@ function sqlQuery($statement, $link)
...
+                                } elseif (($v_acl > $database_acl)) {
+                                    echo "  <td><a href='acl_upgrade.php?site=$sfname' class='text-decoration-none'>Upgrade Access Controls</a></td>\n";
+                                } elseif (($v_realpatch != $database_patch)) {
+                                    echo "  <td><a href='sql_patch.php?site=$sfname' class='text-decoration-none'>Patch Database</a></td>\n";
+                                } else {
+                                    echo "  <td><i class='fa fa-check fa-lg text-success' aria-hidden='true' ></i></a></td>\n";
+                                }
+                                if (($v_database == $database_version) && ($v_acl <= $database_acl) && ($v_realpatch == $database_patch)) {
+                                    echo "  <td><a href='interface/login/login.php?site=$sfname' class='text-decoration-none'><i class='fa fa-sign-in fa-lg' aria-hidden='true' data-toggle='tooltip' data-placement='top' title ='Login to site $sfname'></i></a></td>\n";
```
Comment (bradymiller) [inline admin.php:170]: @stu01509 , There should be a login link here. Note this is not your fault, but is because the fontawesome icon needs to change. need to use `fa-sign-in-alt` instead of `fa-sign-in` above
image

File: sql_upgrade.php [hunk 6]
```diff
@@ -69,101 +69,109 @@
 <html>
 <head>
 <title>OpenEMR Database Upgrade</title>
-<link rel='STYLESHEET' href='interface/themes/style_blue.css'>
-<link rel="shortcut icon" href="public/images/favicon.ico" />
+    <link rel="stylesheet" href="public/assets/bootstrap/dist/css/bootstrap.min.css">
+    <script src="public/assets/jquery/dist/jquery.min.js"></script>
+    <script src="public/assets/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (bradymiller) [inline sql_upgrade.php:74]: hi @stu01509 , Note this script includes globals.php, so should use the Header class for this script.

File: sql_upgrade.php [hunk 7]
```diff
@@ -69,101 +69,109 @@
...
+
+                            echo ">$version</option>\n";
+                        }
+                        ?>
+                    </select>
+                </div>
+                <p>If you are unsure or were using a development version between two
+                    releases, then choose the older of possible releases.</p>
+                <p class="text-danger">If you are upgrading from a version below 5.0.0 to version 5.0.0 or greater, do note that this upgrade can take anywhere from several minutes to several hours (you will only see a whitescreen until it is complete; do not stop the script before it is complete or you risk corrupting your data).</p>
+                <input type='submit' class='btn btn-primary' name='form_submit' value='Upgrade Database' />
```
Comment (bradymiller) [inline sql_upgrade.php:105]: change to button and note since this script includes globals you have access to the fontawesome button icons :)

File: admin.php (hunk 5 above)
Comment (stu01509) [reply admin.php:170]: Thanks for your mention, I fixed it.
It looks like good.

image

File: sql_upgrade.php (hunk 7 above)
Comment (stu01509) [reply sql_upgrade.php:105]: Hi @bradymiller 

I include the `globals.php` file, change the input to button and add the `btn-update`,
but it seems odd, is correct?

image

Comment (bradymiller) [reply sql_upgrade.php:105]: Agreed. Maybe btn-transmit icon. If that looks funny also, and none of the others look right,, then wouldn't use one.

btw, the script already includes globals.php, so you shouldn't need to do that.

**Comment by bradymiller** [comment]
hi @stu01509 , Looks great and testing well, so bringing this into codebase. Thanks for the improvements! -brady
